internal/goproxy/proxy.go

```diff
@@ -15,76 +15,62 @@ import (
 )
 
 type config struct {
-	differentCache bool
-	forceBuild     bool
-	toolexecArgs   string
+	toolexec string
 }
 
 type Option func(*config)
 
-const goCacheVar = "GOCACHE"
-
-// WithForceBuild forces a call to Run() to build all go files and dependencies
-// when wrapping a `go build` command. It effectively adds the `-a` flag to the build
-func WithForceBuild() Option {
-	return func(c *config) {
-		c.forceBuild = true
-	}
-}
-
-// WithDifferentCache forces a call to Run() to override the GOCACHE directory
-// A temporary directory is instead created and all build artifacts are cached there
-func WithDifferentCache() Option {
-	return func(c *config) {
-		c.differentCache = true
-	}
-}
-
 // WithToolexec forces a call to Run() to build with the -toolexec option when
 // wrapping a build command
-func WithToolexec(args []string) Option {
-	var sb strings.Builder
-	return func(c *config) {
-		for _, arg := range args {
-			sb.WriteString(fmt.Sprintf("%s ", arg))
+func WithToolexec(args ...string) Option {
```

Comment:
How about the following since we at least expect the toolexec binary to be provided
```suggestion
func WithToolexec(tool string, args ...string) Option {
```